src/models/invites-ignorer.ts

```diff
@@ -313,10 +349,15 @@ export class IgnoredInvites {
      * Modify in place the `IGNORE_INVITES_POLICIES` object from account data.
      */
     private async withIgnoreInvitesPolicies(cb: (ignoreInvitesPolicies: {[key: string]: any}) => void) {
-        const { policies, ignoreInvitesPolicies } = this.getPoliciesAndIgnoreInvitesPolicies();
-        cb(ignoreInvitesPolicies);
-        policies[IGNORE_INVITES_ACCOUNT_EVENT_KEY.name] = ignoreInvitesPolicies;
-        await this.client.setAccountData(POLICIES_ACCOUNT_EVENT_TYPE.name, policies);
+        await this._withIgnoreInvitesPoliciesLock.acquireAsync();
+        try {
+            const { policies, ignoreInvitesPolicies } = this.getPoliciesAndIgnoreInvitesPolicies();
+            cb(ignoreInvitesPolicies);
+            policies[IGNORE_INVITES_ACCOUNT_EVENT_KEY.name] = ignoreInvitesPolicies;
+            await this.client.setAccountData(POLICIES_ACCOUNT_EVENT_TYPE.name, policies);
+        } finally {
+            this._withIgnoreInvitesPoliciesLock.release();
+        }
```

Comment:
Not sure a lock is better here, as it yield N `setAccountData` calls which don't de-dup lower down so the promise-d/deferred approach probably works better here also

Reply:
I'm not sure what you have in mind.

Here, if the code is called N times, it's because we have N different things to write, so writing N times kinda makes sense. I suppose that I could accumulate them into a single call to `setAccountData` by storing the changes somewhere, to decrease network I/O in the unlikely case `withIgnoreInvitesPolicies` is called concurrently?

This would definitely complicate the code, though, and I don't really see what it optimizes for.

Or am I misunderstanding what you have in mind?

Reply:
This lock is just providing linearisation, which can be accomplished with a Promise. 

```
withIgnoreInvitesPoliciesPromise = Promise.resolve<void>();
...
await this.withIgnoreInvitesPoliciesPromise;
...
this.withIgnoreInvitesPoliciesPromise = this.client.setAccountData(...);
```

Reply:
Also there are missing types in `getPoliciesAndIgnoreInvitesPolicies`

Reply:
> Also there are missing types in `getPoliciesAndIgnoreInvitesPolicies`

Not following. Do you wish me to give names to `{[key: string]: any}` types?